client/src/user-context.tsx

```diff
@@ -18,6 +17,43 @@ export type UserPlusSettings = {
   collectionLastModified: Date | null;
 };
 
+export class OfflineSettingsData {
+  offline?: boolean;
+  preferOnline?: boolean;
+  autoUpdates?: boolean;
```

Comment:
Aren't these properties non-optional?

```suggestion
  offline: boolean;
  preferOnline: boolean;
  autoUpdates: boolean;
```